magefiles/utils.go

```diff
@@ -295,6 +295,46 @@ func cleanupQuayTags(quayService quay.QuayService, organization, repository stri
 	return fmt.Errorf("encountered errors during CleanupQuayTags: %s", errBuilder.String())
 }
 
+func repoNameStartsWithPrefix(prefixes []string, repoName string) bool {
+	for _, prefix := range prefixes {
+		hasPrefix := strings.HasPrefix(repoName, prefix)
+		if hasPrefix {
+			return true
+		}
+	}
+	return false
+}
+
+// Deletes the private repos older than 7 days
+func cleanupPrivateRepos(quayService quay.QuayService, quayOrg string, repoNamePrefixes []string) error {
+	repos, err := quayService.GetAllRepositories(quayOrg)
+	if err != nil {
+		return err
+	}
+	totalPrivateCount := 0
+	deletedPrivateCount := 0
+	for _, repo := range repos {
+		if !repo.IsPublic && repoNameStartsWithPrefix(repoNamePrefixes, repo.Name) {
+			lastModifiedTime := int64(repo.LastModified)
+			currentTime := time.Now().Unix()
+			timeDiff := int64(7 * 24 * 3600)
+			if currentTime-lastModifiedTime > timeDiff {
+				fmt.Printf("Last modified time in human readable format is %v for repo %s\n", time.Unix(int64(repo.LastModified), 0), repo.Name) // For debugging, to be removed later
+				_, err := quayService.DeleteRepository(quayOrg, repo.Name)
+				if err != nil {
+					return fmt.Errorf("failed to delete repository %s with error: %v", repo.Name, err)
+				}
```

Comment:
Maybe the function should try to delete all the repos rather than exiting on the first error? And return an error at the end if it fails to delete any repo.

That way the job can still clean up most repos even if there's an intermittent error for deleting one of the repos.

Reply:
Thanks. Updated it now.